editor/src/com/talosvfx/talos/editor/addons/scene/widgets/directoryview/DirectoryViewWidget.java

```diff
@@ -423,7 +424,57 @@ public void drop (DragAndDrop.Source source, DragAndDrop.Payload payload, float
 
 					openDirectory(targetItem.fileHandle.path());
 				} else {
-					logger.info("TODO COPY INTO DIRECTORY and do asset stuff");
+					if (payloadObject instanceof GlobalDragAndDrop.ArrayDragAndDropPayload) {
+						Array<GlobalDragAndDrop.BaseDragAndDropPayload> items = ((GlobalDragAndDrop.ArrayDragAndDropPayload) payloadObject).getItems();
+						FileHandle lastCopiedFile = null;
+						for (GlobalDragAndDrop.BaseDragAndDropPayload item : items) {
+							if (item instanceof GlobalDragAndDrop.FileHandleDragAndDropPayload) {
+								AssetRepository assetRepository = AssetRepository.getInstance();
+								FileHandle handle = ((GlobalDragAndDrop.FileHandleDragAndDropPayload) item).getHandle();
+								if (fileHandle.child(handle.name()).exists()) {
+									// file is already present, see if it should be replaced
+									String title = "Oh no!";
+									String message = "An older item named \"" + handle.name() + "\" already \n exists in this location. Do you want to replace it with the newer \n one you're moving?";
+									Runnable keep = new Runnable() {
+										@Override
+										public void run() {
+											FileHandle newHandle = assetRepository.copyRawAsset(handle, fileHandle);
+											String projectPath = SharedResources.currentProject.rootProjectDir().path();
+											projectExplorerWidget.loadDirectoryTree(projectPath);
```

Comment:
Why are re reloading the entire widget in this loop every time, just in the current directory?

Reply:
Because when you have conflicts with many files and resolve them you wanna see all the changes in real time.